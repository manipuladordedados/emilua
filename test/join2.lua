-- This test ensures join() propagates values back from the joinee.

local fib = spawn(function()
    print('secondary fiber starts')
    return 30, 31, 32
end)

print('secondary fiber spawned')
local ret = {fib:join()}
print(#ret)
print(ret[1])
print(ret[2])
print(ret[3])
print('end of main fiber')
