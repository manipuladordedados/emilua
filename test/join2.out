secondary fiber spawned
secondary fiber starts
3
30
31
32
end of main fiber
